### PR TITLE
First release

### PR DESCRIPTION
**Pull request by johan-smc**
### Summary
Merge all the complete issues in the project.

### Test Plan
Testing two modules in different cases.

### Related Issues
Not related issues.

### CODE REVIEW

File: src/main/java/tree/ImpTargetTreeNode.kt [hunk 1]
```diff
@@ -1,30 +1,71 @@
...
-        return this
+    fun setParent(newParent: TargetTreeNode?) {
+        parent = newParent
     }
 
-    override val name: String
-        get() = "name"
+    private fun getRightSibling(son: TargetTreeNode): TargetTreeNode? {
+        if (children.isNotEmpty()) {
+            val index = (children as List<ImpTargetTreeNode>).binarySearch(son as ImpTargetTreeNode, ImpTargetTreeNodeComparator())
```
Comment (DorianMoreno) [inline src/main/java/tree/ImpTargetTreeNode.kt]: These casts generate an unchecked cast warning. Should we take care of them? or add a suppress warning?

File: src/main/java/tree/ImpTargetTreeNode.kt [hunk 2]
```diff
@@ -1,30 +1,71 @@
...
+        }
+        return null
+    }
+
+    fun preorder() {
+        println("$name:$tag($id)")
+        for (child in children) {
+            (child as ImpTargetTreeNode).preorder()
+        }
+    }
```
Comment (DorianMoreno) [inline src/main/java/tree/ImpTargetTreeNode.kt]: This function is never used in this release and was previously used as a testing mechanism but we now have more robust ones, so i think it should disappear.
```suggestion
```

**Comment by DorianMoreno** [comment]
On a side note, should we edit the README for our first release?

File: src/main/java/tree/ImpTargetTreeNode.kt (hunk 1 above)
Comment (johan-smc) [reply src/main/java/tree/ImpTargetTreeNode.kt]: Yes I have checked and we need to add a suppress warning

File: src/main/java/tree/ImpTargetTreeNode.kt (hunk 2 above)
Comment (johan-smc) [reply src/main/java/tree/ImpTargetTreeNode.kt]: No, maybe this function will be used after in the project, is better keep it

Comment (DorianMoreno) [reply src/main/java/tree/ImpTargetTreeNode.kt]: Sounds right

**Pull request merged**